src/VolumeCache.ts

```diff
@@ -139,69 +108,35 @@ export default class VolumeCache {
     this.last = this.last.prev;
   }
 
-  /** Evicts a specific entry from the cache */
-  // TODO use this to intelligently evict redundant data
+  /** Evicts a specific entry from the cache. */
   private evict(entry: CacheEntry): void {
     this.removeEntryFromStore(entry);
     this.removeEntryFromList(entry);
   }
 
   /**
-   * Prepares a new cached volume with the specified channels, times, and scales.
-   * @returns {CacheStore} A container for cache entries for this volume.
-   * A `CacheStore` may only be accessed or modified by passing it to this class's methods.
-   */
-  public addVolume(numChannels: number, numTimes: number, scaleDims: Vector3[]): CacheStore {
-    const makeTCArray = (): CacheEntry[][][] => {
-      const tArr: CacheEntry[][][] = [];
-      for (let i = 0; i < numTimes; i++) {
-        const cArr: CacheEntry[][] = [];
-        for (let j = 0; j < numChannels; j++) {
-          cArr.push([]);
-        }
-        tArr.push(cArr);
-      }
-      return tArr;
-    };
-
-    const scales = scaleDims.map((size): CacheStoreMultiscaleLevel => ({ size, data: makeTCArray() }));
-    return { scales, numTimes, numChannels };
-  }
-
-  /**
-   * Add a new array to the cache (representing a subset of a channel's extent at a given time and scale)
-   * @returns {boolean} a boolean indicating whether the insertion succeeded
+   * Add a new entry to the cache.
+   * @returns {boolean} a boolean indicating whether the insertion succeeded.
    */
-  public insert(volume: CacheStore, data: ArrayBuffer, optDims: Partial<DataArrayExtent> = {}): boolean {
-    const scaleCache = volume.scales[optDims.scale || 0];
-    const entryList = scaleCache.data[optDims.time || 0][optDims.channel || 0];
-    const chunk = optDims.chunk ?? new Vector3(0, 0, 0);
-    const { size } = scaleCache;
-
-    if (chunk.x < 0 || chunk.y < 0 || chunk.z < 0 || chunk.x >= size.x || chunk.y >= size.y || chunk.z >= size.z) {
-      console.error("VolumeCache: attempt to insert into nonexistent chunk");
-      return false;
-    }
+  public insert(key: string, data: ArrayBuffer): boolean {
     if (data.byteLength > this.maxSize) {
       console.error("VolumeCache: attempt to insert a single entry larger than the cache");
       return false;
     }
 
     // Check if entry is already in cache
-    for (const existingEntry of entryList) {
-      if (existingEntry.chunk.equals(chunk)) {
-        existingEntry.data = data;
-        this.moveEntryToFirst(existingEntry);
-        return true;
-      }
+    // This will move the entry to the front of the LRU list, if present
```

Comment:
Not suggesting you change this but it could be interesting if we needed an option to not do this. Like getEntry(key, doNotUpdateUsage)  My thought is that maybe you want to update LRU only on usage, not on insertion.  Totally hypothetical.  I suppose that will be more apparent in the future tickets.